### DISCUSSION TITLE
Zusammenhang NRF24 + CMT2300A + Display

### DISCUSSION THREAD

**Discussion by harryinet** [tbnobody/OpenDTU#2231] [Q&A]
Hi, ich habe HM-2250-6T Wechselrichter. Von einem Bekannten eine DTU, weil dieser sein Kennwort vergessen hatte.
Wegen 4MB die Version 24.08.18 per Kabel aufgespielt.
Einigen Profiles probiert, ich kann den HM-WR aber nur erkennen, wenn NRF24 + CMT2300A aktiviert sind.
Zu guter Letzt noch das Display 1309 dazu aktiviert dann passt es.

Mein älterer HM-1500 (4x PV) wir nie erkannt.

Wo finde ich etwas über diese Zusammenhänge?

Der Radio Chip ist E49 900M20S von eByte

Danke fürs Lesen und ggf. antworten.

**Comment by ghost** [comment]
Hi,

du musst genauer werden ;)

Die Firmware v24.08.18 gibt es nicht, aber eine v24.8.1.
Du kannst auch die aktuelle v24.8.5. verwenden, die v24.8.1. hat Fehler.

Dann ist ein HM-2250-6T nicht in der Kompatibilitätsliste verzeichnet. Ist es ein HMT-2250-6T?
Bei letzterem würde das mit dem CMT2300A ( E49 900M20S) Funkmodul passen. Daher klappt es nur mit dem Profil, wo das CMT mit drin steht.

Für den WechselrichterHM-1500-4T wird ein anderes / weiteres Funkmodul Typ NRF24L01+ benötigt. ([sowas](https://www.ebay.de/itm/404310175373?chn=ps&_ul=DE&norover=1&mkevt=1&mkrid=707-168914-873919-9&mkcid=2&itemid=404310175373&targetid=325425753764&device=c&mktype=pla&googleloc=9043015&poi=1029203&campaignid=20352335475&mkgroupid=152759828764&rlsatarget=pla-325425753764&abcId=&merchantid=5077338099&gad_source=1&gclid=Cj0KCQjwz7C2BhDkARIsAA_SZKY-h5o7bgUayuPGgiMyy5FVO-sSz40dsd7cBefDISDwOiEzCVkW_tgaAtJhEALw_wcB) oder [sowas](https://eckstein-shop.de/NRF24L0124GHzAntennaSPIInterfaceWirelessTransceiverEmpfC3A4ngerFunkModulforArduino?ws_oss_lieferland=DE&gad_source=1&gclid=Cj0KCQjwz7C2BhDkARIsAA_SZKb519YIrznHAYD-l-Q-9RxOtCfSetgBHypT7CFodbOKfWMaqYAISCUaAl8bEALw_wcB), wobei ich das 1. einsetze.) Wichtig ist das "+" in der Bezeichnung beim Bestellen. Das müsste nachgerüstet werden. Kannst du mal ein Foto vom Innenleben deiner DTU hochladen? Dann könnten wir hier beurteilen, ob du das noch brauchst.

Welches Funkmodul für welchen Wechselrichter benötigt wird steht auf der [Startseite](https://github.com/tbnobody/OpenDTU), musst etwas nach unten scrollen.  Dokumentation ist [hier](https://www.opendtu.solar/) zu finden.

**Reply by ghost** [reply]
Ach ja:

Deine tatsächlich angeschlossenen Hardware muss der openDTU bekannt gemacht werden.
Das geschieht per Pinkonfiguration (Pin Mapping) an Hand eines [Device Profile](https://www.opendtu.solar/firmware/device_profiles/#structure-of-the-json-file).
Je nach dem, [welche Firmware](https://www.opendtu.solar/firmware/flash_esp/) du einsetzt, ist es erforderlich, eine [Pin-Konfigurationsdatei](https://github.com/tbnobody/OpenDTU/tree/master/docs/DeviceProfiles) auf die openDTU hochzuladen.
Firmware ohne ESP32 im Namen (mit Kommentar: deprecated! /veraltet) benötigt keine Pin-Mapping Datei, alle anderen schon.

In den Links hier im Text kannst du dir den Aufbau der Device Profiles ansehen und dir dein eigenes basteln.

**Reply by harryinet** [reply]
Danke für euer Interesse, ich habe also:

- Diese Firmware wegen 4MB https://github.com/helgeerbe/OpenDTU-OnBattery/releases/tag/2024.08.18
- Es ist auch nur der Chip CMT2300A als Hukepack auf dem ESP32, NRF24 kann ich nicht erkennen
- Es ist der HMT-2250-6T 

Ich hatte diverse Profile ausprobiert und mit einem ging schon mal Display und CMT aber including NFR24. Das war mehr Zufall und rumprobieren.

Vorhin habe ich es etwas zusammengeschrumpft ...
```
  [
      {
        "name": "CMT2300A MIT SSD1309",
        "nrf24": {
            "miso": -1,
            "mosi": -1,
            "clk": -1,
            "irq": -1,
            "en": -1,
            "cs": -1
        },
        "cmt": {
            "clk": 12,
            "sdio": 14,
            "cs": 27,
            "fcs": 26,
            "gpio2": -1,
            "gpio3": -1
        },
        "eth": {
            "enabled": false
        },
        "display": {
            "type": 4,
            "data": 21,
            "clk": 22
        }
    }
]
```

Da nun so alles funktioniert ist es super.

**Reply by ghost** [reply]
klingt gut, so sollte das Display und das CMT funktionieren.
Ohne NRF24 wird allerdings dein HM 1500 nicht gefunden. das müsstest du noch einbauen.

**Reply by harryinet** [reply]
Das oben beschriebene Modell ist ist si Okay, für meine persönliche Anlage habe ich openDTUfusion seinerzeit beschafft, also kein Problem damit. Sehr cool ... https://www.opendtu.solar/3rd_party/opendtu_fusion/
